### PR TITLE
VideoPress: reload block on rating change

### PR DESCRIPTION
**Pull request by thedebian**
Fixes Automattic/greenhouse#1005

#### Changes proposed in this Pull Request:

When updating the rating on a VideoPress video block, the block is now reloaded to reflect the changes (for the age gate in particular).
This PR also fixes the meta API that was previously checking the existing rating instead of the new one.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* On a Jetpack site with VideoPress enabled
* Edit or create a post with a /video block
* Upload or select a video from your media library already uploaded on VideoPress
* In the block settings, play with the "Rating" option
Capture d’écran 2021-11-05 à 11 51 17
* ✅ The block should be regenerated every time the "rating" option is changed
* ✅ The block should display the age gate if "R" is selected
* ✅ The block should not display the age gate if anything else is selected
* After selecting a rating, reload the page and select the video block
* ✅ The previously selected rating should be displayed

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ public function videopress_block_update_meta( $request ) {
 					$should_update_meta                  = true;
 				}
 
-				if ( isset( $json_params['rating'] ) && isset( $meta['videopress']['rating'] ) && videopress_is_valid_video_rating( $meta['videopress']['rating'] ) ) {
+				if ( isset( $json_params['rating'] ) && isset( $meta['videopress']['rating'] ) && videopress_is_valid_video_rating( $json_params['rating'] ) ) {
```
Comment (jgcaruso) [inline projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php:147]: Nice catch, the previous check doesn't seem to make much sense!

**Review by jgcaruso — CHANGES_REQUESTED** [review]
This tested well, but there is a strange sidebar refresh happening that I'm wondering if we can address?

When the rating is changed, it looks like the video options are being duplicated. I *think* it might be because the original block's sidebar options are being rendered along with the existing ones while it decides what should actually be rendered? I've attached a video.


https://user-images.githubusercontent.com/4081020/140763254-a9400cf4-4dfd-4156-afbe-04b5b6425e17.mov

**Comment by thedebian** [comment]
@jgcaruso 
> When the rating is changed, it looks like the video options are being duplicated. I _think_ it might be because the original block's sidebar options are being rendered along with the existing ones while it decides what should actually be rendered? 

I think this bug has been introduced when we stopped removing the CoreVideoEdit block from the React tree to make the noticeOperations work, this render is tied to the loading state and, while the CoreVideoEdit block gets hidden, the InspectorControls are not.
By looking at the Gutenberg video/edit.js, it seems there are no options to not render the InspectorControls with this block.
We may be able to find a workaround but it doesn't seem related to this PR, I'll create a new UX friction issue in Greenhouse to address it.

**Review by jgcaruso — APPROVED** [review]

**Review by samiff — APPROVED** [review]

**Comment by thedebian** [comment]
Hey @samiff , I had to rebase this PR (and added a bit of mess in between before force-pushing to solve it), could you re-check if everything is still ok and re-approve please ?

**Review by samiff — APPROVED** [review]

**Comment by thedebian** [comment]
Changeset 235109

**Pull request merged**